thirdparty/breakpad/mainwindow.cpp

```diff
@@ -0,0 +1,334 @@
+    str2.empty();
+
+    mystr = &str1;
+
+    for (i = 0; i < line.size(); i++){
+            if ( line[i] == ',' && myflag == false){
+                    mystr = &str2;
+                    myflag = true;
+            }
+            else{
+                    if (line[i] != '\n'){
+                            *mystr += line[i];
+                    }
+
+            }
+    }
+
+    return pair<string,string>(str1,str2);
+
+}
+
+QMap <QString,QString> read_csv(QString mypath){
+        string line;
+        pair<string,string> mykeyval;
+        ifstream myfile(mypath.toStdString());
+        QMap <QString,QString> mymap;
+
+        if (myfile.is_open()){
+                while ( getline (myfile,line) ){
+                        cout << line << '\n';
+                        mykeyval = line2strings(line);
+                        //cout << "str1: " << mykeyval.first << " str2: " << mykeyval.second << endl;
+                        mymap.insert(mykeyval.first.c_str(),mykeyval.second.c_str());
+
+                }
+                myfile.close();
+
+        }
+        else{
+            cout << "Unable to open file";
+        }
+
+        return mymap;
+}
+
+
+MainWindow::MainWindow(QWidget *parent) :
+    QMainWindow(parent),
+    ui(new Ui::MainWindow)
+{
+    ui->setupUi(this);
+    m_manager = new QNetworkAccessManager(this);
+    connect(m_manager, &QNetworkAccessManager::finished, this, &MainWindow::uploadFinished);
+
+}
+
+MainWindow::~MainWindow()
+{
+    delete ui;
+}
+
+void MainWindow::sslErrors(const QList<QSslError> &sslErrors)
+{
+#ifndef QT_NO_SSL
+    foreach (const QSslError &error, sslErrors)
+        fprintf(stderr, "SSL error: %s\n", qPrintable(error.errorString()));
+#else
+    Q_UNUSED(sslErrors);
+#endif
+}
+
+void MainWindow::uploadFinished(QNetworkReply *reply)
+{
+    if (!reply->error())
+    {
+        m_file->close();
+        m_file->deleteLater();
+        reply->deleteLater();
+    }
+}
+
+void MainWindow::onError(QNetworkReply::NetworkError err)
+{
+    qDebug() << " SOME ERROR!";
+    qDebug() << err;
+}
+
+void MainWindow::sendReportQt(QString user_txt){
+    QString minidump_path;
+    QString metadata_path;
+
+    if ( QCoreApplication::arguments().count() == 3 ){
+
+        minidump_path = QCoreApplication::arguments().at(1);
+        metadata_path = QCoreApplication::arguments().at(2);
+
+        QStringList filePathList = minidump_path.split('/');
+        QString minidump_filename = filePathList.at(filePathList.count() - 1);
+
+        qDebug() << "minidump file: " << minidump_path;
+
+        QString url = "https://musescore.sp.backtrace.io:6098/post?format=minidump&token=00268871877ba102d69a23a8e713fff9700acf65999b1f043ec09c5c253b9c03";
+
+        QString boundary = "--"
+                    + QString::number(
+                            qrand() * (90000000000) / (RAND_MAX + 1) + 10000000000, 16);
+
+        QNetworkRequest request(url);
+        request.setHeader(QNetworkRequest::ContentTypeHeader,
```

Comment:
Add a comment here to explain that the boundary needs to be added in the header too.